File: src/java/main/commands/RichJson_invoke.java
package commands;

import core.RichJsonCommand;
import core.RichJsonContext;
import core.RichJsonParser;

import java.util.function.Supplier;

public class RichJson_invoke implements RichJsonCommand {
    @Override
    public Object execute(RichJsonParser parser, RichJsonContext context) throws Exception {
        if (context.currentMember instanceof Supplier) {
            return ((Supplier<?>) context.currentMember).get();
        }
        throw new Exception(String.format("RichJson the given function in '%s' is not a supplier", context.currentName));
    }
}

File: src/c#/main/commands/RichJson_invoke.cs
using System;
using main.core;

namespace main.commands
{
    public class RichJson_invoke
    {
        public object Execute(RichJsonParser parser, RichJsonContext context)
        {
            if (context.CurrentMember is Func<object> supplier)
            {
                return supplier();
            }
            throw new Exception($"RichJson the given function in '{context.CurrentName}' is not a supplier");
        }
    }
}